recipe_scrapers/instructables.py

```diff
@@ -0,0 +1,49 @@
+from ._abstract import AbstractScraper
+from ._grouping_utils import group_ingredients
+from ._utils import normalize_string
+
+
+class Instructables(AbstractScraper):
+    @classmethod
+    def host(cls):
+        return "instructables.com"
+
+    def ingredient_groups(self):
+        return group_ingredients(
+            self.ingredients(),
+            self.soup,
+            ".step-body p strong",
+            ".step-body li",
+        )
+
+    def title(self):
+        title_tag = self.soup.find("h2", class_="step-title sr-only")
+        return normalize_string(title_tag.get_text())
+
+    def author(self):
+        author_tag = self.soup.find("span", class_="author-name")
+        return normalize_string(author_tag.get_text().replace("By ", ""))
+
+    def ingredients(self):
+        ingredients_section = self.soup.find("section", id="stepsupplies")
+        if not ingredients_section:
+            return []
+
+        ingredients_tags = ingredients_section.find_all("li", class_="ql-indent-1")
+        return [normalize_string(tag.get_text()) for tag in ingredients_tags]
+
+    def instructions(self):
+        steps = self.soup.find_all("section", class_="step")
+        instructions = []
+        for step in steps:
+            step_title = step.find("h2", class_="step-title")
+            step_body = step.find("div", class_="step-body")
+            if step_title and step_body:
+                instructions.append(
+                    f"{normalize_string(step_title.get_text())}: {normalize_string(step_body.get_text())}"
+                )
```

Comment:
For this scraper, since the `Step n: ...` text is somewhat descriptive (not simply `step 1`, `step 2`, ...), I think that the results might be more comprehensible with separate instruction entries for those.

e.g. instead of `["Step 1: Preparation: do some things", "Step 2: Begin cooking: some more things"]`, I would suggest `["Step 1: Preparation", "Do some things", "Step 2: Begin cooking", "some more things"].

The way I think about this is that I imagine the text of the instructions being spoken aloud by cooking assistant software that allows scrolling up/down by pressing the arrow keys.

I'd find it annoying/distracting to scroll up and hear `step 3`, because that wouldn't tell me anything about what the step is or whether I'd found the part of the instructions I'm looking for.  In contrast, `step 3: combine the sauces` could be useful.